### ISSUE TITLE
Move Operation helper types to Azure.Core

### ISSUE THREAD

**Issue by annelo-msft** [Azure/autorest.csharp#2231]
We need to move the following files out of generator shared source in the autorest.csharp repo into Azure.Core shared source in the azure-sdk-for-net repo:

- IOperationSource
- LowLevelFuncOperation
- LowLevelOperationHelpers
- NextLinkOperationImplementation
- OperationFinalStateVia
- OperationInternals
- OperationInternalsOfT
- OperationOrResponseInternals
- OperationOrResponseInternalsOfT
- HttpPipelineExtensions

This is for two reasons:
- We need to be able to modify operation types without being blocked by generator merge to repo processes
- We need to consolidate all the operation types in a single place so we can begin refactoring and improving the LRO types as a unified story

**Comment by annelo-msft** [comment]
@AlexanderSher, @m-nash, do you have any concerns about this?

**Comment by ShivangiReja** [comment]
There is an [issue ](https://github.com/Azure/azure-sdk-for-net/issues/27483) to move HttpPipelineExtensions methods to Azure.Core. Can we do this as a part of this issue?

**Comment by AlexanderSher** [comment]
These types should be just removed:
```cs
OperationInternals
OperationInternalsOfT
OperationOrResponseInternals
OperationOrResponseInternalsOfT
```

**Comment by annelo-msft** [comment]
@AlexanderSher, by removed, do you mean deleted from both repos?

**Comment by AlexanderSher** [comment]
We should also try to get rid of `IOperationSource`, but that is more complicated thing to do.

**Comment by annelo-msft** [comment]
Duplicate of https://github.com/Azure/azure-sdk-for-net/issues/28369
